src/sickle.ts

```diff
@@ -144,106 +148,109 @@ class Annotator {
     this.currentNode = node;
     // this.logWithIndent('node: ' + ts.SyntaxKind[node.kind]);
     this.indent++;
-    switch (node.kind) {
-      case ts.SyntaxKind.ModuleDeclaration:
-        if (node.flags & ts.NodeFlags.Ambient) {
-          // An ambient module declaration only declares types for TypeScript's
-          // benefit, so we want to skip all Sickle processing of it.
+
+    if (node.flags & ts.NodeFlags.Ambient) {
+      this.visitExterns(node);
+      // An ambient declaration declares types for TypeScript's benefit, so we want to skip Sickle
+      // conversion of its contents.
+      this.writeRange(node.getFullStart(), node.getEnd());
+    } else {
+      switch (node.kind) {
+        case ts.SyntaxKind.InterfaceDeclaration:
+          let decl = <ts.InterfaceDeclaration>node;
           this.writeRange(node.getFullStart(), node.getEnd());
           break;
-        } else {
+        case ts.SyntaxKind.VariableDeclaration:
+          this.maybeEmitJSDocType((<ts.VariableDeclaration>node).type);
           this.writeNode(node);
-        }
-        break;
-      case ts.SyntaxKind.VariableDeclaration:
-        this.maybeEmitJSDocType((<ts.VariableDeclaration>node).type);
-        this.writeNode(node);
-        break;
-      case ts.SyntaxKind.ClassDeclaration:
-        let classNode = <ts.ClassDeclaration>node;
-        if (classNode.members.length > 0) {
-          // We must visit all members individually, to strip out any
-          // /** @export */ annotations that show up in the constructor
-          // and to annotate methods.
-          this.writeTextBetween(classNode, classNode.members[0]);
-          for (let member of classNode.members) {
-            this.visit(member);
+          break;
+        case ts.SyntaxKind.ClassDeclaration:
+          let classNode = <ts.ClassDeclaration>node;
+          if (classNode.members.length > 0) {
+            // We must visit all members individually, to strip out any
+            // /** @export */ annotations that show up in the constructor
```

Comment:
All of this green code here is just indented one more level.  I considered refactoring to an additional function to prevent this indenting.  The only real change to this function is the above check for NodeFlags.Ambient, which shuttles this whole piece of the AST off to a helper function instead of doing this processing.